public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -0,0 +1,660 @@
+  const details = receivers.find((receiver) => receiver.name === contactPoint);
+  if (!details) {
+    return (
+      <Link to={createContactPointLink(contactPoint, alertManagerSourceName)}>
+        <Strong>{contactPoint}</Strong>
+      </Link>
+    );
+  }
+
+  const integrations = details.grafana_managed_receiver_configs;
+  if (!integrations) {
+    return (
+      <Link to={createContactPointLink(contactPoint, alertManagerSourceName)}>
+        <Strong>{contactPoint}</Strong>
+      </Link>
+    );
+  }
+
+  const groupedIntegrations = groupBy(details.grafana_managed_receiver_configs, (config) => config.type);
+
+  return (
+    <HoverCard
+      arrow
+      placement="top"
+      header={
+        <MetaText icon="at">
+          <div>Contact Point</div>
+          <Strong>{contactPoint}</Strong>
+        </MetaText>
+      }
+      key={uniqueId()}
+      content={
+        <Stack direction="row" gap={0.5}>
+          {/* use "label" to indicate how many of that type we have in the contact point */}
+          {Object.entries(groupedIntegrations).map(([type, integrations]) => (
+            <Label
+              key={uniqueId()}
+              label={integrations.length > 1 ? integrations.length : undefined}
+              icon={INTEGRATION_ICONS[type]}
+              value={upperFirst(type)}
+            />
+          ))}
+        </Stack>
+      }
+    >
+      <Link to={createContactPointLink(contactPoint, alertManagerSourceName)}>
+        <Strong>{contactPoint}</Strong>
+      </Link>
+    </HoverCard>
+  );
+};
+
+const wildcardRouteWarning = <Badge icon="exclamation-triangle" text="Matches all labels" color="orange" />;
+
+function getContactPointErrors(contactPoint: string, contactPointsState: ReceiversState): JSX.Element[] {
+  const notifierStates = Object.entries(contactPointsState[contactPoint]?.notifiers ?? []);
+  const contactPointErrors = notifierStates.reduce((acc: JSX.Element[] = [], [_, notifierStatuses]) => {
+    const notifierErrors = notifierStatuses
+      .filter((status) => status.lastNotifyAttemptError)
+      .map((status) => (
+        <Label
+          icon="at"
+          key={uniqueId()}
+          label={`Contact Point › ${status.name}`}
+          value={status.lastNotifyAttemptError}
+        />
+      ));
+
+    return acc.concat(notifierErrors);
+  }, []);
+
+  return contactPointErrors;
+}
+
+const routePropertyToLabel = (key: keyof InhertitableProperties): string => {
+  switch (key) {
+    case 'receiver':
+      return 'Contact Point';
+    case 'group_by':
+      return 'Group by';
+    case 'group_interval':
+      return 'Group interval';
+    case 'group_wait':
+      return 'Group wait';
+    case 'mute_time_intervals':
+      return 'Mute timings';
+    case 'repeat_interval':
+      return 'Repeat interval';
+  }
+};
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  matcher: (label: string) => {
+    const { color, borderColor } = getTagColorsFromName(label);
```

Comment:
Won't css performance suffer from this approach?

Reply:
Yea it might, I believe this is negligible though and I'm not quite sure how to improve this – will extracting this in to its own function impact the performance of rendering the other styles / classNames?